### PR TITLE
Add support for loongarch

### PR DESCRIPTION
**Pull request by loongson-sm**
Hello Maintainers,

I've submitted a pull request  that adds support for the loongrch architecture. 
I've thoroughly tested this PR on a loongrch-based system, and it appears to work as expected without any adverse effects on other architectures. 
If you have any questions, concerns, or suggestions regarding this PR, please don't hesitate to let me know. I'm open to making any necessary adjustments to ensure the successful integration of this feature into the project.
Thank you for your time and consideration.

Best regards,
Meng Sang

### CODE REVIEW

File: build/cpu.py [hunk 1]
```diff
@@ -43,6 +43,11 @@ class IA64(CPU):
 	'''
 	name = 'ia64'
 
+class LOONGARCH64(CPU):
```
Comment (mthuurne) [inline build/cpu.py]:
```suggestion
class LoongArch64(CPU):
```
A web search suggests that "LoongArch" is the most common capitalization and it fits with the CamelCase typically used in Python class names.

File: build/cpu.py [hunk 2]
```diff
@@ -43,6 +43,11 @@ class IA64(CPU):
...
+   '''64-bit LOONGARCH64.
+   '''
+   name = 'loongarch64'
```
Comment (mthuurne) [inline build/cpu.py]: It seems the indentation for these lines is different from the rest of the file. Please indent with one tab.

(While 4 spaces is more common in Python, tabs are what the C++ part of openMSX uses and we stuck with that for internal consistency.)

**Review by mthuurne — CHANGES_REQUESTED** [review]
Thanks for your PR! The change itself looks fine to me, only some cosmetic changes are needed.

File: build/cpu.py (hunk 2 above)
Comment (loongson-sm) [reply build/cpu.py]: Sure, I'll make some modifications. Thank you for your suggestions.

Comment (mthuurne) [reply build/cpu.py]: Thanks!

**Comment by mthuurne** [comment]
I cherry-picked your commit to the `master` branch: d77fe49007a72594b273cd6077537de6e6d77118. It seems something went wrong with the indentation, but I amended that. The CI is failing probably because of an expired certificate, but that is unrelated to this PR.

Thanks for your contribution!

**Comment by loongson-sm** [comment]
I really appreciate your help and the time you took to review and merge the changes.
If you have any suggestions or feedback for me in the future, please don't hesitate to share.

**Pull request closed without merging**